.circleci/config.yml

```diff
@@ -309,12 +309,33 @@ commands:
             # Add agent source code to PYTHONPATH to be able to import its components.
             export PYTHONPATH=.:${PYTHONPATH}
             # generate file with checksum that is calculated for every required file.
-            python -m tests.distribution_builders.<< parameters.distribution >>.cmd --checksum > ~/checksum
+            python -m tests.image_builder.distributions.<< parameters.distribution >>.cmd --checksum > ~/checksum
+            python -m tests.image_builder.distributions.fpm_package_builder.cmd --checksum > ~/fpm_builder_checksum
+
+      - restore_cache:
+          # here the 'tar' file for fpm builder image is stored. Use it to restore it without building.
+          key: fpm-builder-{{ .Branch }}-{{ checksum "~/fpm_builder_checksum" }}-<< pipeline.parameters.cache_version >>
 
       - restore_cache:
           # here the 'tar' files of the docker images is stored. Use it to restore agent image without building it.
           key: smoke-distribution-image-{{ .Branch }}-<< parameters.distribution >>-{{ checksum "~/checksum" }}-<< pipeline.parameters.cache_version >>
 
+
+      - run:
+          name: Copy cached images if they exist.
+          command: |
+            mkdir -p ~/agent_image
```

Comment:
I think it would be better to move this into a script file in scripts/circle/ directory or similar.

I also plan to do this in the near future for some of the benchmark targets, which currently contain tons of in-line commands where a shell script file would be a better fit.